datalad/support/archives.py

```diff
@@ -142,6 +142,23 @@ def decompress_file(archive, dir_, leading_directories='strip'):
         if cmo.err:
             lgr.debug("patool gave stderr:\n%s" % cmo.err)
 
+    # Note: (ben) Experienced issue, where extracted tarball
+    # lacked execution bit of directories, leading to not being
```

Comment:
do you have an example?  could there be  a tiny unittest for such a case?

Reply:
Well, I have huge dicom tarballs that have that issue. But I don't know how to create them on purpose, since I only consume them.
The only way to test would be to mock that extracting result and see that the added code is actually setting the X bit. But then: What we really need to test for would be sideeffects and this is basically done by not breaking other tests, no?

Reply:
Correct. I just thought that may be there is no mystery how this tarball came about. My wild guess it's that it was done programmatically, and then some bits, like TarInfo.mode if was done in python, were not set properly 😉

Reply:
FWIW: It's done on Windows, so this mystery could possibly involve all kinds of shit. ;-)

Reply:
Sounds like a time to think about adding appveyor setup to run at least subset of tests of windows (e.g. eliminating those which we know to fail with direct mode etc).  #2083 .  Then I guess we could have observed/tested that behavior